components/EditableTitle.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from "react";
+import styles from "../layouts/default.layout.module.scss";
+
+export const EditableTitle = (props) => {
+  const { defaultValue, readOnly, onConfirm } = props;
+  const [projectTitle, setProjectTitle] = useState(defaultValue);
+
+  const handleConfirm = (e) => {
+    if (e.code === "Enter") {
+      e.preventDefault();
+      e.target.blur();
+    }
+  };
+
+  useEffect(() => {
+    setProjectTitle(defaultValue);
+  }, [defaultValue]);
+
+  return (
+    <input
+      value={projectTitle}
+      readOnly={readOnly}
+      className={styles.projectName}
+      style={{ width: projectTitle.length + "ch" }}
```

Comment:
Is the Equinor font a monospace font?

Ref `ch` description: `Relative to the width of the "0" (zero)`. So for monospace fonts this would be correct, otherwise you might end up with clipped content.